web/scripts/generateZonesConfig.ts

```diff
@@ -40,29 +40,45 @@ const getConfig = (): CombinedZonesConfig => {
   ]);
 
   const contributors = new Set<string>();
+  const zones: Record<string, ZoneConfig> = {};
 
-  const zones = filesWithDirectory.reduce((zones, filepath) => {
+  for (const filepath of filesWithDirectory) {
     const config = yaml.load(fs.readFileSync(filepath, 'utf8')) as ZoneConfig;
 
-    if (config.contributors) {
-      for (const contributor of config.contributors) {
-        contributors.add(contributor);
-        const index = config.contributors?.indexOf(contributor);
-        const contributorArray = [...contributors];
-        const globalIndex = contributorArray.indexOf(contributor);
-        config.contributors
-          ? ((config as unknown as OptimizedZoneConfig).contributors[index] = globalIndex)
-          : [];
+    const zoneContributors = new Set<string>();
+
+    for (const subZoneName of config.subZoneNames ?? []) {
+      const subZonePath = `${basePath}/${subZoneName}.yaml`;
+      const subZoneConfig = yaml.load(fs.readFileSync(subZonePath, 'utf8')) as ZoneConfig;
+      if (subZoneConfig.contributors) {
+        for (const contributor of subZoneConfig.contributors) {
+          contributors.add(contributor);
+          zoneContributors.add(contributor);
+        }
```

Comment:
We should consider making this async in the future as for example the US has a lot of sub zones that could be processed in async. However this would make it none deterministic and more complicated so I opted against it.

Reply:
I had an idea that would solve and speed this up, will mark as draft and refractor.